engine/src/org/pentaho/di/trans/steps/mappinginput/MappingInput.java

```diff
@@ -165,21 +176,6 @@ public boolean processRow( StepMetaInterface smi, StepDataInterface sdi ) throws
     return true;
   }
 
-  private void replaceRenameIfNecessary( RowMetaInterface rowMeta, MappingValueRename valueRename )
```

Comment:
Sorry, I didn't notice there was a revert and a new commit mixed into the same PR.
It still seems like it would be worth removing the duplicated code here, but I notice that there is a very slight change to the extracted method's code where it does an additional check, so reverting first is probably the right thing.

Okay, this looks good to me from a review standpoint.  I'll run through the tests in just a bit.